jbmc/src/jbmc/jbmc_parse_options.cpp

```diff
@@ -249,6 +249,22 @@ void jbmc_parse_optionst::get_command_line_options(optionst &options)
   if(cmdline.isset("string-printable"))
     options.set_option("string-printable", true);
 
+  if(cmdline.isset("no-refine-strings") && cmdline.isset("string-printable"))
+  {
+    error() << "--string-printable cannot be used together with "
+            << "--no-refine-strings" << eom;
+    exit(CPROVER_EXIT_USAGE_ERROR);
+  }
+
+  if(
+    cmdline.isset("no-refine-strings") &&
+    cmdline.isset("max-nondet-string-length"))
+  {
+    error() << "--max-nondet-string-length cannot be used together with "
+            << "--no-refine-strings" << eom;
+    exit(CPROVER_EXIT_USAGE_ERROR);
+  }
```

Comment:
I don't think such checks are practical. If I want to do a run of test-gen without string refinement, I don't want to have to go through all string-related options to remove them.

Reply:
@owen-jones-diffblue, any opinions?

Reply:
I think option combinations that have no effect should not be accepted.

Reply:
How about a warning message?

Reply:
ok, let's go for that compromise